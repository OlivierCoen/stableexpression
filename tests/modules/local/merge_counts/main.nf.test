nextflow_process {

    name "Test Process MERGE_COUNTS"
    script "modules/local/merge_counts/main.nf"
    process "MERGE_COUNTS"
    tag "merge"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file('$projectDir/tests/input/merge_counts/counts1.csv')
                input[1] = file('$projectDir/tests/input/merge_counts/counts2.csv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with(process.out.merged) {
                assert path(get(0)).readLines().size() == 6
            }
        }

    }

}
