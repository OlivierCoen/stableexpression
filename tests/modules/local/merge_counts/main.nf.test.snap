{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "merged.csv:md5,d8e0472f9704eaf615052ae68f9544e8"
                ],
                "1": [
                    [
                        "MERGE_COUNTS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "MERGE_COUNTS",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "merged": [
                    "merged.csv:md5,d8e0472f9704eaf615052ae68f9544e8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-02T13:30:33.87202502"
    }
}